### ISSUE TITLE
Bug: Truth tracking example broken by recent commit

### ISSUE THREAD

**Issue by gagnonlg** [acts-project/acts#636]
Commit 95b21bcc2cfdcdeafb6784e6020122048d5a4ba5 introduced the following lines in Core/include/Acts/Propagator/Navigator.hpp:

```
std::vector<GeometryIdentifier> externalSurfaces;
if (!state.navigation.externalSurfaces.empty()) {
  auto layerID =
      state.navigation.navLayerIter->object->geometryId().layer();
  auto externalSurfaceRange =
      state.navigation.externalSurfaces.equal_range(layerID);

  externalSurfaces.reserve(
      state.navigation.externalSurfaces.count(layerID));
  for (auto itSurface = externalSurfaceRange.first;
       itSurface != externalSurfaceRange.second; itSurface++) {
    externalSurfaces.push_back(itSurface->second);
  }
}
```

While running the [RecTruthTracks example](https://acts.readthedocs.io/en/latest/howto/run_truth_tracking.html ), the code segfaults at the `auto layerID =
      state.navigation.navLayerIter->object->geometryId().layer();` line. Poking around in gdb, it seems that at the time of the call `navLayerIter` is an iterator into an empty vector so there is an illegal dereference.

I've tried the parent commit (094764295209220a1fa7747202109ae340aae612) and the code works fine there.

Steps to reproduce:

$BUILD/bin/ActsExamplePythia8 \
    --events=100 \
    --output-dir=data/gen/ttbar_mu140 \
    --output-csv=1 \
    --rnd-seed=42 \
    --gen-cms-energy-gev=14000 \
    --gen-hard-process=Top:qqbar2ttbar=on \
    --gen-npileup=140 \
    --jobs 6

$BUILD/bin/ActsExampleFatrasGeneric \
    --input-dir=data/gen/ttbar_mu140 \
    --output-dir=data/sim_generic/ttbar_mu140 \
    --output-csv=1 \
    --select-eta=-3:3 \
    --select-pt=0.5: \
    --remove-neutral \
    --bf-value=0 0 2 \
    --digi-geometric-3d \
    --jobs 6

$BUILD/bin/ActsExampleTruthTracksGeneric \
    --input-dir=data/sim_generic/ttbar_mu140 \
    --bf-value=0 0 2 \
    --output-dir=data/reco_generic/ttbar_mu140 \
    --jobs 6

 I attach [backtrace.txt](https://github.com/acts-project/acts/files/5712597/backtrace.txt) in case it's useful.

**Comment by Corentin-Allaire** [comment]
Hi, I will have a look this morning.

**Comment by Corentin-Allaire** [comment]
@gagnonlg I don't have Pythia installed locally, could you send me directly your input files for  ActsExampleTruthTracksGeneric ?

**Comment by Corentin-Allaire** [comment]
This should be fixed by #640 but I cannot test it. Could someone try it ?

**Comment by gagnonlg** [comment]
Hello @Corentin-Allaire, 

Thanks for the prompt response. 


I will try the fix later today and get back to you. Meanwhile, I attach one of the pythia input files ([test-data.tar.gz](https://github.com/acts-project/acts/files/5717207/test-data.tar.gz)) which should be sufficient to trigger the bug.

**Comment by asalzburger** [comment]
> Hello @Corentin-Allaire,
> 
> Thanks for the prompt response.
> 
> I will try the fix later today and get back to you. Meanwhile, I attach one of the pythia input files ([test-data.tar.gz](https://github.com/acts-project/acts/files/5717207/test-data.tar.gz)) which should be sufficient to trigger the bug.

Thanks, appreciated.

**Comment by Corentin-Allaire** [comment]
Thanks ! I will probably not have the time tonight but tomorrow morning I can try to look at the bug in more detail.

**Comment by gagnonlg** [comment]
@Corentin-Allaire I cab confirm that #640 fixes the problem on my end, thanks!

**Comment by Corentin-Allaire** [comment]
This was resolved with #640
